x-pack/plugins/security_solution/public/management/components/console/service/parse_command_input.test.ts

```diff
@@ -5,141 +5,164 @@
-  ): ParsedCommandInterface => {
-    return {
-      input: '',
-      name: 'foo',
-      args: {},
-      hasArgs: Object.keys(overrides.args || {}).length > 0,
-      ...overrides,
-    } as ParsedCommandInterface;
-  };
-
-  it.each([
-    ['foo', 'foo'],
-    ['    foo', 'foo'],
-    ['    foo     ', 'foo'],
-    ['foo     ', 'foo'],
-    [' foo-bar  ', 'foo-bar'],
-  ])('should identify the command entered: [%s]', (input, expectedCommandName) => {
-    const parsedCommand = parseCommandInput(input);
-
-    expect(parsedCommand).toEqual(
-      parsedCommandWith({
-        input,
-        name: expectedCommandName,
+import {
+  parseCommandInput,
+  ParsedCommandInterface,
+  parsedPidOrEntityIdParameter,
+} from './parsed_command_input';
+
+describe('when using parsed command input utils', () => {
+  describe('when using parseCommandInput()', () => {
+    const parsedCommandWith = (
+      overrides: Partial<Omit<ParsedCommandInterface, 'hasArg' | 'hasArgs'>> = {}
+    ): ParsedCommandInterface => {
+      return {
+        input: '',
+        name: 'foo',
         args: {},
-      })
-    );
+        hasArgs: Object.keys(overrides.args || {}).length > 0,
+        ...overrides,
+      } as ParsedCommandInterface;
+    };
+
+    it.each([
+      ['foo', 'foo'],
+      ['    foo', 'foo'],
+      ['    foo     ', 'foo'],
+      ['foo     ', 'foo'],
+      [' foo-bar  ', 'foo-bar'],
+    ])('should identify the command entered: [%s]', (input, expectedCommandName) => {
+      const parsedCommand = parseCommandInput(input);
+
+      expect(parsedCommand).toEqual(
+        parsedCommandWith({
+          input,
+          name: expectedCommandName,
+          args: {},
+        })
+      );
+    });
+
+    it('should parse arguments that the `--` prefix', () => {
+      const input = 'foo    --one     --two';
+      const parsedCommand = parseCommandInput(input);
+
+      expect(parsedCommand).toEqual(
+        parsedCommandWith({
+          input,
+          args: {
+            one: [],
+            two: [],
+          },
+        })
+      );
+    });
+
+    it('should parse arguments that have a single string value', () => {
+      const input = 'foo --one value --two=value2';
+      const parsedCommand = parseCommandInput(input);
+
+      expect(parsedCommand).toEqual(
+        parsedCommandWith({
+          input,
+          args: {
+            one: ['value'],
+            two: ['value2'],
+          },
+        })
+      );
+    });
+
+    it('should parse arguments that have multiple strings as the value', () => {
+      const input = 'foo --one value for one here --two=some more strings for 2';
+      const parsedCommand = parseCommandInput(input);
+
+      expect(parsedCommand).toEqual(
+        parsedCommandWith({
+          input,
+          args: {
+            one: ['value for one here'],
+            two: ['some more strings for 2'],
+          },
+        })
```

Comment:
I maintain that _this specific behavior_ is weird and atypical of  *any* terminal-like interface. And if we ship anything that lets a user rely on it, we've done something wrong. This seems to just be a spacing fix, but seeing _tests making sure a behavior not common to text-based consoles is present_ is stirring the red flags

Reply:
@pzl 

Chatted offline with @paul-tavares and @caitlinbetz 

We are going to revisit the parser when it comes to this concern.  We will refactor the logic to require `""` for multi word inputs.  For example, `--comment "this is a comment"` will be required